### PR TITLE
magento/magento2-login-as-customer#180: Deprecated records should be removed from a login_as_customer table by cron.

### PR DESCRIPTION
**Pull request by engcom-Foxtrot**
### Fixed Issues (if relevant)

1. Fixes magento/magento2-login-as-customer#180: Deprecated records should be removed from login_as_table by cron.

### Manual testing scenarios (*)

1. Perform Login as Customer action.
2. Make sure some records are present in the login_as_customer table.
3. Run cron.
4. Wait.
5. Make sure no records older than 60 seconds are present in the login_as_customer table.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed
**Result:**
Peek 2020-06-23 14-32

File: app/code/Magento/LoginAsCustomer/Model/ResourceModel/DeleteAuthenticationData.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+namespace Magento\LoginAsCustomer\Model\ResourceModel;
+
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\Stdlib\DateTime\DateTime;
+use Magento\LoginAsCustomerApi\Api\ConfigInterface;
+
+/**
+ * Delete deprecated authentication data.
+ */
+class DeleteAuthenticationData
```
Comment (naydav) [inline app/code/Magento/LoginAsCustomer/Model/ResourceModel/DeleteAuthenticationData.php:17]: DeleteExpiredAuthenticationData

File: app/code/Magento/LoginAsCustomer/Model/ResourceModel/DeleteAuthenticationData.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    /**
+     * Delete deprecated authentication data.
+     */
+    public function execute(): void
+    {
+        $connection = $this->resourceConnection->getConnection();
+        $tableName = $this->resourceConnection->getTableName('login_as_customer');
+
+        $timePoint = date(
```
Comment (naydav) [inline app/code/Magento/LoginAsCustomer/Model/ResourceModel/DeleteAuthenticationData.php:57]: Please use `new \DateTime($date, new \DateTimeZone('UTC'))`

File: app/code/Magento/LoginAsCustomer/Model/ResourceModel/DeleteAuthenticationData.php (hunk 1 above)
Comment (naydav) [reply app/code/Magento/LoginAsCustomer/Model/ResourceModel/DeleteAuthenticationData.php:17]: Please add interface

Comment (engcom-Foxtrot) [reply app/code/Magento/LoginAsCustomer/Model/ResourceModel/DeleteAuthenticationData.php:17]: @naydav done.

File: app/code/Magento/LoginAsCustomer/Model/ResourceModel/DeleteAuthenticationData.php (hunk 2 above)
Comment (engcom-Foxtrot) [reply app/code/Magento/LoginAsCustomer/Model/ResourceModel/DeleteAuthenticationData.php:57]: @naydav done.

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed
Result as same as in [comment](app/code/Magento/LoginAsCustomer/etc/crontab.xml)

**Comment by naydav** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
Issue no longer valid.

**Pull request closed without merging**